### PR TITLE
Some doctests/note for essentials

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Review by StefanKarpinski — APPROVED** [review]

File: base/essentials.jl [hunk 1]
```diff
@@ -417,9 +430,11 @@ esc(@nospecialize(e)) = Expr(:escape, e)
 
 Annotates the expression `blk` as a bounds checking block, allowing it to be elided by [`@inbounds`](@ref).
 
-Note that the function in which `@boundscheck` is written must be inlined into
-its caller with [`@inline`](@ref) in order for `@inbounds` to have effect.
+!!! note
+    The function in which `@boundscheck` is written must be inlined into
+    its caller with [`@inline`](@ref) in order for `@inbounds` to have effect.
```
Comment (fredrikekre) [inline base/essentials.jl]: Is this really true? If the compiler inlines the function without `@inline`, won't the `@boundscheck` block be elided?

Comment (StefanKarpinski) [reply base/essentials.jl]: Correct, this claim is wrong: inlining for any reason is sufficient.

Comment (kshyatt) [reply base/essentials.jl]: Should I just delete this `!!! note` and everything indented then?

Comment (StefanKarpinski) [reply base/essentials.jl]: Just delete the `with [@inline](@ref)` part.

**Comment by kshyatt** [comment]
good to go?

**Pull request merged**